src/libstd/sys/unix/os.rs

```diff
@@ -102,14 +102,19 @@ pub fn error_string(errno: i32) -> String {
 }
 
 pub fn getcwd() -> io::Result<PathBuf> {
-    let mut buf = [0 as c_char; BUF_BYTES];
-    unsafe {
-        if libc::getcwd(buf.as_mut_ptr(), buf.len() as libc::size_t).is_null() {
-            Err(io::Error::last_os_error())
-        } else {
-            Ok(bytes2path(CStr::from_ptr(buf.as_ptr()).to_bytes()))
+    super::fill_bytes_buf(|buf, len| {
+        unsafe {
+            Some(if !libc::getcwd(buf, len).is_null() {
+                Ok(bytes2path(CStr::from_ptr(buf).to_bytes()))
+            } else {
+                let error = io::Error::last_os_error();
+                if error.raw_os_error().unwrap() == libc::ERANGE {
```

Comment:
It's a program error if `raw_os_error` returns `None`, so the unwrap doesn't hurt here IMO.